### PR TITLE
Refactoring to work with modified onos-topo API

### PR DESCRIPTION
**Pull request by tomikazi**

### CODE REVIEW

File: pkg/device/device.go [hunk 1]
```diff
@@ -124,85 +128,110 @@ func (x ListResponseType) String() string {
...
-		ID:         topo.ID(device.ID),
-		Revision:   device.Revision,
-		Attributes: device.Attributes,
-		Type:       topo.Object_ENTITY,
-		Obj: &topo.Object_Entity{
-			Entity: &topo.Entity{
-				KindID:    topo.ID(device.Type),
-				Protocols: device.Protocols,
+	o := device.Object
+	if o == nil {
```
Comment (adibrastegarnia) [inline pkg/device/device.go:134]: Why this condition is needed?

**Review by adibrastegarnia — APPROVED** [review]

**Comment by adibrastegarnia** [comment]
If you can run integration tests then would be great @tomikazi

File: pkg/device/device.go (hunk 1 above)
Comment (tomikazi) [reply pkg/device/device.go:134]: This is in case the original device object were synthesized from scratch.

**Comment by tomikazi** [comment]
> If you can run integration tests then would be great @tomikazi

This will take some doing and will have to wait until the CLI is done as well. Can't do it before then.

**Comment by tomikazi** [comment]
retest this please

**Pull request merged**